pyomo/contrib/parmest/parmest.py

```diff
@@ -613,18 +1181,234 @@ def _Q_opt(
+        independent and follow a Gaussian distribution with constant variance.
+
+        The formula used in parmest was verified against equations (7-5-15) and
+        (7-5-16) in "Nonlinear Parameter Estimation", Y. Bard, 1974.
+
+        This formula is also applicable if the objective is scaled by a constant;
+        the constant cancels out. (was scaled by 1/n because it computes an
+        expected value.)
+        """
+        # check if the user-supplied covariance method is supported
+        try:
+            cov_method = CovarianceMethodLib(method)
+        except ValueError:
+            raise ValueError(
+                f"Invalid method: '{method}'. Choose "
+                f"from {[e.value for e in CovarianceMethodLib]}."
+            )
+
+        # check if the user specified 'SSE' or 'SSE_weighted' as the objective function
+        if self.obj_function == ObjectiveLib.SSE:
+            # check if the user defined the 'measurement_error' attribute
+            if hasattr(model, "measurement_error"):
+                # get the measurement errors
+                meas_error = [
+                    model.measurement_error[y_hat]
+                    for y_hat, y in model.experiment_outputs.items()
+                ]
+
+                # check if the user supplied the values of the measurement errors
+                if all(item is None for item in meas_error):
+                    measurement_var = sse / (
+                        n - l
+                    )  # estimate of the measurement variance
+                    if cov_method == CovarianceMethodLib.reduced_hessian:
+                        cov = (
+                            2 * measurement_var * self.inv_red_hes
+                        )  # covariance matrix
+                        cov = pd.DataFrame(
+                            cov,
+                            index=self.estimated_theta.keys(),
+                            columns=self.estimated_theta.keys(),
+                        )
+                    elif (
+                        cov_method == CovarianceMethodLib.finite_difference
+                        or cov_method
+                        == CovarianceMethodLib.automatic_differentiation_kaug
+                    ):
+                        cov = compute_covariance_matrix(
+                            self.exp_list,
+                            method,
+                            theta_vals=self.estimated_theta,
+                            solver=solver,
+                            step=step,
+                            tee=self.tee,
+                            estimated_var=measurement_var,
+                        )
+                    else:
+                        raise NotImplementedError(
+                            'Only "finite_difference", "reduced_hessian", and '
+                            '"automatic_differentiation_kaug" methods are supported.'
+                        )
```

Comment:
This is unreachable code. You already check in lines 1274-1280 whether or not a supported method was supplied, and it raises an error if not. So this is redundant.